### PR TITLE
ZEN-30929: Add "Model Change Handler time" measurement to RM ZenPack

### PR DESCRIPTION
**Pull request by drevak**
Added the Metrology timers to check the execution of commit hooks.

### CODE REVIEW

**Comment by drevak** [comment]
retest this

File: Products/ZenMessaging/queuemessaging/publisher.py [hunk 1]
```diff
@@ -31,6 +31,8 @@
 log = logging.getLogger('zen.queuepublisher')
 
 MODEL_TYPE = ProtobufEnum(modelevents_pb2.ModelEvent, 'model_type')
+BEFORE_COMMIT_HOOK_TIMER = Metrology.utilization_timer("zen.queuepublisher.beforeCompletionHookTimer")
```
Comment (cluther) [inline Products/ZenMessaging/queuemessaging/publisher.py]: I don't think these meters should be created in the module's global scope. This leads to them being created in process that don't even have access to ZODB. See what happens when I put a pdb.set_trace() on the same line, then run zenpython.

```
(zenoss) [zenoss@5b5f49ae7072 ~]$ zenpython run
> /opt/zenoss/Products/ZenMessaging/queuemessaging/publisher.py(40)<module>()
-> class ModelChangePublisher(object):
(Pdb)
```

If you want them in the module's scope, you could set them to None here, then initialize them on demand when they're used.

File: Products/ZenMessaging/queuemessaging/publisher.py [hunk 2]
```diff
@@ -232,8 +234,10 @@ def getModelChangePublisher():
         tx._synchronizedPublisher = ModelChangePublisher()
         # create new PublishSynchronizer also add after completion hook so that client/channel can be closed
         pSync = PublishSynchronizer()
-        tx.addBeforeCommitHook(pSync.beforeCompletionHook, [tx])
-        tx.addAfterCommitHook(pSync.afterCompletionHook, [tx])
+        with BEFORE_COMMIT_HOOK_TIMER:
+            tx.addBeforeCommitHook(pSync.beforeCompletionHook, [tx])
```
Comment (cluther) [inline Products/ZenMessaging/queuemessaging/publisher.py]: I believe this only causes us to measure how long it takes to add the hook. Not how long the hooks take to execute on each commit.

File: Products/ZenMessaging/queuemessaging/publisher.py (hunk 1 above)
Comment (drevak) [reply Products/ZenMessaging/queuemessaging/publisher.py]: Added the Metrology timers as a decorator to the appropriate methods.
The Metrology package has his own registry package so we only can use it without writing additional getter function.

File: Products/ZenMessaging/queuemessaging/publisher.py (hunk 2 above)
Comment (drevak) [reply Products/ZenMessaging/queuemessaging/publisher.py]: Added the timers as a decorator to `beforeCompletionHook` and `afterCompletionHook` functions.

**Pull request merged**